src/app/clusters/thread-network-diagnostics-server/thread-network-diagnostics-provider.cpp

```diff
@@ -0,0 +1,861 @@
+ */
+CHIP_ERROR WriteThreadNetworkDiagnosticAttributeToTlv(AttributeId attributeId, app::AttributeValueEncoder & encoder)
+{
+    CHIP_ERROR err;
+#if (CHIP_DEVICE_CONFIG_ENABLE_THREAD && !CHIP_DEVICE_CONFIG_USES_OTBR_POSIX_DBUS_STACK)
+    otInstance * otInst = ThreadStackMgrImpl().OTInstance();
+
+    if (!otDatasetIsCommissioned(otInst))
+    {
+        // For the following nullable attributes of the cluster, encodeNull since
+        // thread instance cannot provide the related data when it is not currently configured.
+        //
+        // Note that RoutingRole is nullable but not listed here as thread provides
+        // valid data even when disabled or detached
+        switch (attributeId)
+        {
+        case Attributes::Channel::Id:
+        case Attributes::NetworkName::Id:
+        case Attributes::PanId::Id:
+        case Attributes::ExtendedPanId::Id:
+        case Attributes::MeshLocalPrefix::Id:
+        case Attributes::PartitionId::Id:
+        case Attributes::Weighting::Id:
+        case Attributes::DataVersion::Id:
+        case Attributes::StableDataVersion::Id:
+        case Attributes::LeaderRouterId::Id:
+        case Attributes::ActiveTimestamp::Id:
+        case Attributes::PendingTimestamp::Id:
+        case Attributes::Delay::Id:
+        case Attributes::SecurityPolicy::Id:
+        case Attributes::ChannelPage0Mask::Id:
+        case Attributes::OperationalDatasetComponents::Id:
+            return encoder.EncodeNull();
+        }
+    }
+
+    switch (attributeId)
+    {
+    case Attributes::Channel::Id: {
+        uint16_t channel = otLinkGetChannel(otInst);
+        err              = encoder.Encode(channel);
+    }
+    break;
+
+    case Attributes::RoutingRole::Id: {
+        using ThreadNetworkDiagnostics::RoutingRoleEnum;
+        RoutingRoleEnum routingRole;
+        otDeviceRole otRole = otThreadGetDeviceRole(otInst);
+
+        if (otRole == OT_DEVICE_ROLE_DISABLED)
+        {
+            routingRole = RoutingRoleEnum::kUnspecified;
+        }
+        else if (otRole == OT_DEVICE_ROLE_DETACHED)
+        {
+            routingRole = RoutingRoleEnum::kUnassigned;
+        }
+        else if (otRole == OT_DEVICE_ROLE_ROUTER)
+        {
+            routingRole = RoutingRoleEnum::kRouter;
+        }
+        else if (otRole == OT_DEVICE_ROLE_LEADER)
+        {
+            routingRole = RoutingRoleEnum::kLeader;
+        }
+        else if (otRole == OT_DEVICE_ROLE_CHILD)
+        {
+            otLinkModeConfig linkMode = otThreadGetLinkMode(otInst);
+
+            if (linkMode.mRxOnWhenIdle)
+            {
+                routingRole = RoutingRoleEnum::kEndDevice;
+#if CHIP_DEVICE_CONFIG_THREAD_FTD
+                if (otThreadIsRouterEligible(otInst))
+                {
+                    routingRole = RoutingRoleEnum::kReed;
+                }
+#endif
+            }
+            else
+            {
+                routingRole = RoutingRoleEnum::kSleepyEndDevice;
+            }
+        }
```

Comment:
would a switch-case be clearer?

Reply:
I rather not make a switch case in a switch case. This section of code was already like this in the older function